.github/workflows/ci.yaml

```diff
@@ -12,96 +12,173 @@ defaults:
+            is-draft-pr: true
+    runs-on: ${{ matrix.os }}
 
-      - name: Generate Environment variable docs
-        run: poetry run pre-commit run gen-docs-env-vars --all-files --show-diff-on-failure
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
 
-      - name: Generate pipeline definitions
-        run: poetry run pre-commit run gen-docs-components --all-files --show-diff-on-failure
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          version: ${{ env.UV_VERSION }}
 
       - name: Test
-        run: poetry run pytest tests
+        run: uv run --frozen pytest tests
 
-      - name: Install docs dependencies
-        run: poetry install --with docs
+  docs:
+    name: Docs
+    runs-on: [ubuntu-24.04]
 
-      - name: Check markdown, toml, css formatting
-        uses: dprint/check@v2.2
-        if: runner.os == 'Linux'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: "3.11"
+          version: ${{ env.UV_VERSION }}
+
+      - name: Generate schema (kpops schema)
+        run: uv run --frozen pre-commit run gen-schema --all-files --show-diff-on-failure
+
+      - name: Generate CLI Usage docs (typer-cli)
+        run: uv run --frozen pre-commit run gen-docs-cli --all-files --show-diff-on-failure
+
+      - name: Generate Environment variable docs
+        run: uv run --frozen pre-commit run gen-docs-env-vars --all-files --show-diff-on-failure
+
+      - name: Generate pipeline definitions
+        run: uv run --frozen pre-commit run gen-docs-components --all-files --show-diff-on-failure
 
       - name: Test docs build (mkdocs)
-        run: poetry run mkdocs build -f docs/mkdocs.yml
+        run: uv run --frozen --group=docs mkdocs build -f docs/mkdocs.yml
 
   publish-snapshot-version:
     name: Publish snapshot to TestPyPI
-    needs: [test]
-    uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-snapshot.yaml@1.40.4
-    with:
-      python-version: "3.11"
-      poetry-version: "1.7.1"
-    secrets:
-      pypi-token: ${{ secrets.TEST_PYPI_TOKEN }}
+    needs: [lint, test, docs]
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: "3.11"
+          version: ${{ env.UV_VERSION }}
+
+      - name: Generate snapshot time
+        id: snapshot-time
+        run: |
+          snapshot_time=$(python -c "from datetime import datetime; time = datetime.now(); print(time.strftime('%Y%m%d%H%M%S'))")
+          echo "snapshot-time=${snapshot_time}" >> "$GITHUB_OUTPUT"
+
+      - name: Bump version patch
+        id: bump-version
+        uses: bakdata/ci-templates/actions/bump-version@1.50.4
+        with:
+          release-type: patch
+
+      - run: git restore .
```

Comment:
Wait, so you are bumping and then restoring? Are you pushing in between or what is the purpose of doing this?

Reply:
this is needed otherwise bumpversion (in the configuration we use) will complain that the Git working directory is unclean. We only need the new version result after the first `bumpversion patch` call here which is stored in a step output

Reply:
I've added a comment

Reply:
Shouldn't `--allow-dirty` do the trick?

Reply:
it might, but I don't see this input parameter for the bump-version action. I could use a hacky workaround and pass it as `release-type: patch --allow-dirty` or we could add it but personally I don't mind the `git restore` here.